mm2src/coins/lp_coins.rs

```diff
@@ -2150,15 +2159,48 @@ impl CoinsContext {
         })))
     }
 
-    pub async fn add_coin(&self, coin: MmCoinEnum) -> Result<(), MmError<CoinIsAlreadyActivatedErr>> {
+    pub async fn add_token(&self, coin: MmCoinEnum) -> Result<(), MmError<CoinIsAlreadyActivatedErr>> {
+        let mut coins = self.coins.lock().await;
+        if coins.contains_key(coin.ticker()) {
+            return MmError::err(CoinIsAlreadyActivatedErr {
+                ticker: coin.ticker().into(),
+            });
+        }
+        let ticker = coin.ticker();
+
+        let mut platform_coin_tokens = self.platform_coin_tokens.lock();
+        //        let pc = coin.pla
+        // Here, we tried to add to a token to platform_coin_tokens if the token belongs to a platform coin.
+        if let Some(platform) = platform_coin_tokens.get_mut(coin.platform_ticker()) {
+            platform.insert(ticker.to_owned());
+        }
+
+        coins.insert(ticker.into(), coin);
+        Ok(())
+    }
+
+    pub async fn add_l2(&self, coin: MmCoinEnum) -> Result<(), MmError<CoinIsAlreadyActivatedErr>> {
         let mut coins = self.coins.lock().await;
         if coins.contains_key(coin.ticker()) {
             return MmError::err(CoinIsAlreadyActivatedErr {
                 ticker: coin.ticker().into(),
             });
         }
+        let ticker = coin.ticker().to_string();
+        let platform_ticker = coin.platform_ticker().to_string();
+
+        let mut platform_coin_tokens = self.platform_coin_tokens.lock();
+        if let Some(tokens) = platform_coin_tokens.get_mut(&platform_ticker) {
+            if !tokens.contains(&ticker) {
+                tokens.insert(ticker.clone());
+            };
+        } else {
+            let mut data = HashSet::with_capacity(1);
+            data.insert(ticker.clone());
+            platform_coin_tokens.insert(platform_ticker, data);
```

Comment:
This doesn't work, we need to insert platform coin `LightningCoin` on activation else tokens e.g `tBTC-TEST-lightning` is not going to be picked up at all when deactivating platform coin. So we need to this
```rust
if let Some(tokens) = platform_coin_tokens.get_mut(&platform_ticker) {
             if !tokens.contains(&ticker) {
             tokens.insert(ticker.clone());
         } else {
             let mut data = HashSet::with_capacity(1);
             data.insert(ticker.clone());
             platform_coin_tokens.insert(platform_ticker, data);
```

instead of just this in `add_token` 
```rust
if let Some(platform) = platform_coin_tokens.get_mut(coin.platform_ticker()) {
            platform.insert(ticker.to_owned());
        }
  ```

Reply:
Yes, so this is why I suggested https://github.com/KomodoPlatform/atomicDEX-API/pull/1525#discussion_r1032423977
In a nutshell, you don't push the ticker to `platform_coin_tokens` on `lp_register_coin`. I suggest this:
```rust
async fn lp_register_coin(coin: MmCoinEnum, ...) {
  let mut coins = coins_ctx.coins.lock();
  coins.insert(coin.ticker().to_string(), coin);
  if coin.ticker() == coin.platform_ticker() {
    let mut platform_coin_tokens = coins_ctx.platform_coin_tokens.lock();
    match platform_coin_tokens.entry(coin.ticker().to_string()) {
      Entry::Vacant(entry) => entry.insert(Vec::new()),
      Entry::Occupied(_) => todo!() // probably we should remove the coin from `CoinsContext::coins` container AND return an error, but on the other hand we could just don't insert it.
    }
  }
}
```

Reply:
"but on the other hand we could just don't insert it"